src/components/scenes/FioNameConfirmScene.js

```diff
@@ -0,0 +1,263 @@
+import { type Theme, type ThemeProps, cacheStyles, withTheme } from '../services/ThemeContext.js'
+
+export type LocalState = {
+  balance: number | null,
+  sliderDisabled: boolean | false,
+  loading: boolean | false
+}
+
+export type StateProps = {
+  fioPlugin: EdgeCurrencyConfig | null,
+  denominationMultiplier: string,
+  isConnected: boolean
+}
+
+export type NavigationProps = {
+  fioName: string,
+  paymentWallet: EdgeCurrencyWallet,
+  fee: number,
+  ownerPublicKey: string
+}
+
+type Props = NavigationProps & StateProps & ThemeProps
+
+class FioNameConfirm extends React.Component<Props, LocalState> {
+  state: LocalState = {
+    balance: null,
+    sliderDisabled: false,
+    loading: false
+  }
+
+  componentDidMount() {
+    this.setBalance()
+  }
+
+  isFioAddress = () => {
+    const { fioName } = this.props
+    return fioName.indexOf(Constants.FIO_ADDRESS_DELIMITER) > -1
+  }
+
+  toggleButton = () => {
+    const { fee } = this.props
+    const { balance } = this.state
+    if (balance !== null) {
+      if (fee > balance) {
+        this.setState({
+          sliderDisabled: true
+        })
+      }
+    }
+  }
+
+  setBalance = async () => {
+    const { paymentWallet } = this.props
+    try {
+      const balance = await paymentWallet.getBalance()
+
+      if (balance != null) {
+        const newBalance = parseFloat(truncateDecimals(bns.div(balance, this.props.denominationMultiplier, 18), 6))
+        this.setState({
+          balance: newBalance
+        })
+      }
+
+      this.toggleButton()
+    } catch (e) {
+      this.setState({
+        balance: 0
+      })
+    }
+  }
+
+  saveFioName = async () => {
+    const { isConnected, fioName, paymentWallet, fioPlugin, ownerPublicKey, fee } = this.props
+    if (!isConnected) {
+      showError(s.strings.fio_network_alert_text)
+      return
+    }
+
+    this.setState({ loading: true })
+
+    if (!fee) {
+      if (this.isFioAddress()) {
+        try {
+          if (!fioPlugin) {
+            throw new Error(s.strings.fio_register_address_err_msg)
+          }
+          const response = await fioPlugin.otherMethods.buyAddressRequest(
+            {
+              address: fioName,
+              referralCode: fioPlugin.currencyInfo.defaultSettings.defaultRef,
+              publicKey: ownerPublicKey
+            },
+            true
+          )
+          if (response.error) {
+            if (response.error === 'You have already registered a free address for that domain' && response.code === 400) {
+              const publicDomains = await fioPlugin.otherMethods.getDomains(fioPlugin.currencyInfo.defaultSettings.fallbackRef)
+              const domainExists = publicDomains.find(domain => domain.domain === fioName.split(Constants.FIO_ADDRESS_DELIMITER)[1])
+              if (domainExists && !domainExists.free) {
+                Alert.alert(s.strings.fio_address_register_pay_title, s.strings.fio_address_register_pay, [{ text: s.strings.string_ok_cap }])
```

Comment:
Do not use the system alert. Use an airship modal instead. ButtonModal is ideal and easy to use.